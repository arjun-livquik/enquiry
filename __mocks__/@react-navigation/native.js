export const BackHandler = jest.fn();
export const NavigationContainer = ({children}) => children;
