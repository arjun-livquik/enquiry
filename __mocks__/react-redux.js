export const connect = (mapStateToProps, mapDispatchToProps) => (
  ReactComponent,
) => ReactComponent;

export const Provider = ({children}) => children;

export const useDispatch = () => {};
