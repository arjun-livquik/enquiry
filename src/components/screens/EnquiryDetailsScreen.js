// @flow

import React from 'react';
import {connect} from 'react-redux';
import {View, Text, StyleSheet} from 'react-native';
import {callUser} from '../../config/util';
import {ScrollView, TouchableOpacity} from 'react-native-gesture-handler';
import IconButton from '../IconButton';
import PhoneIcon from '../svg/PhoneIcon';
import {Avatar} from '../Avatar';
import Title from '../Title';
import stringify from 'json-stable-stringify';
const styles = StyleSheet.create({
  container: {flex: 1},
  itemWrapper: {
    backgroundColor: '#d4b5b0',
    borderBottomColor: 'rgba(0,0,0, .2)',
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },
  listItemWrapper: {
    backgroundColor: '#fff',
    borderBottomColor: 'rgba(0,0,0, .2)',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    paddingHorizontal: 16,
  },
  infoWrapper: {marginVertical: 16, flex: 1, justifyContent: 'center'},
  actionWrapper: {margin: 16, flex: 1},
});

type Item = {
  enqId: number,
  name: string,
  isStarted: Boolean,
  phoneNumber: stringify,
  isFree: boolean,
  categoryName: string,
  classLocPref: string,
  createdOn: string,
  location: string,
  postedOn: string,
  providerType: string,
  tag: string,
};

type Props = {
  item: Item,
};

const EnquiryDetailsScreen: (Props) => React$Node = (props) => {
  const {item} = props;
  const {name, enqId, isStarted, phoneNumber, isFree} = item;
  const displayData = [
    {name: 'location', value: 'Location'},
    {name: 'categoryName', value: 'Category'},
    {name: 'classLocPref', value: 'Class Location Preference'},
    {name: 'createdOn', value: 'Created On'},
    {name: 'postedOn', value: 'Posted On'},
    {name: 'providerType', value: 'Provider'},
    {name: 'tag', value: 'Tag'},
  ];
  return (
    <View style={styles.container}>
      <View style={styles.itemWrapper}>
        <Avatar>{name}</Avatar>
        <View style={styles.infoWrapper}>
          <Title>{name}</Title>
        </View>
      </View>
      <View style={styles.itemWrapper}>
        <View style={styles.actionWrapper}>
          <Title>Enquiry Id: {enqId}</Title>
          <Text>{isStarted ? '' : 'Not'} Started</Text>
          <Text>{isFree ? 'Free' : 'Paid'}</Text>
        </View>
        <IconButton onPress={callUser(phoneNumber)}>
          <PhoneIcon width={32} height={32} />
        </IconButton>
      </View>
      <ScrollView>
        <TouchableOpacity
          style={styles.listItemWrapper}
          onPress={callUser(phoneNumber)}>
          <View style={styles.infoWrapper}>
            <Title>Phone Number</Title>
            <Text>{phoneNumber}</Text>
          </View>
        </TouchableOpacity>
        {displayData.map(({name: key, value}) => (
          <View style={styles.listItemWrapper} key={key}>
            <View style={styles.infoWrapper}>
              <Title>{value}</Title>
              <Text>{item[key]}</Text>
            </View>
          </View>
        ))}
      </ScrollView>
    </View>
  );
};

const mapStateToProps = (state) => {
  return {
    item: state?.enquiryReducer?.item,
  };
};

export default connect(mapStateToProps)(EnquiryDetailsScreen);
