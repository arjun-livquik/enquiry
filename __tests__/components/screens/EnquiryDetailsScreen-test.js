import React from 'react';
import renderer from 'react-test-renderer';
import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import EnquiryDetailsScreen from '../../../src/components/screens/EnquiryDetailsScreen';

const mockStore = configureStore([]);

describe('EnquiryDetailsScreen Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  const name = 'Arjun';
  const enqId = '4567';
  const isStarted = false;
  const phoneNumber = '+919878989898';
  const isFree = true;
  const location = 'Home';
  it('Should compare snapshot', () => {
    let root;
    renderer.act(() => {
      root = renderer.create(
        <Provider store={store}>
          <NavigationContainer>
            <EnquiryDetailsScreen
              item={{
                name,
                enqId,
                isStarted,
                phoneNumber,
                isFree,
                location,
                categoryName: '',
                classLocPref: '',
                createdOn: '',
                postedOn: '',
                tag: '',
              }}
            />
          </NavigationContainer>
        </Provider>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
