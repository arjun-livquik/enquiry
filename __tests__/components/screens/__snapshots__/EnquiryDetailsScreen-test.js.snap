// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnquiryDetailsScreen Component Should compare snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#d4b5b0",
        "borderBottomColor": "rgba(0,0,0, .2)",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#9266c9",
          "borderRadius": 50,
          "height": 40,
          "justifyContent": "center",
          "margin": 16,
          "width": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
          }
        }
      >
        A
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Arjun
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#d4b5b0",
        "borderBottomColor": "rgba(0,0,0, .2)",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "margin": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Enquiry Id: 
        4567
      </Text>
      <Text>
        Not
         Started
      </Text>
      <Text>
        Free
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 50,
          "height": 40,
          "justifyContent": "center",
          "margin": 16,
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={32}
        bbWidth={32}
        focusable={false}
        height={32}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 32,
              "width": 32,
            },
          ]
        }
        vbHeight={512}
        vbWidth={512}
        width={32}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M352 0H160c-22.08.026-39.974 17.92-40 40v432c.026 22.08 17.92 39.974 40 40h192c22.08-.026 39.974-17.92 40-40V40c-.026-22.08-17.92-39.974-40-40zm24 472c0 13.255-10.745 24-24 24H160c-13.255 0-24-10.745-24-24v-48h240v48zm0-64H136V88h240v320zm0-336H136V40c0-13.255 10.745-24 24-24h192c13.255 0 24 10.745 24 24v32z"
            fill={4278190080}
            propList={
              Array [
                "fill",
                "stroke",
              ]
            }
            stroke={4278190080}
          />
          <RNSVGPath
            d="M256 432c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24zm0 32a8 8 0 110-16 8 8 0 010 16zM288 40h-64a8 8 0 000 16h64a8 8 0 000-16zM352 376h-8a8 8 0 000 16h8a8 8 0 000-16zM320 376h-8a8 8 0 000 16h8a8 8 0 000-16zM288 376h-8a8 8 0 000 16h8a8 8 0 000-16z"
            fill={4278190080}
            propList={
              Array [
                "fill",
                "stroke",
              ]
            }
            stroke={4278190080}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Phone Number
          </Text>
          <Text>
            +919878989898
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Location
          </Text>
          <Text>
            Home
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Category
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Class Location Preference
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Created On
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Posted On
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Provider
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "rgba(0,0,0, .2)",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Tag
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
